DirectProgramming/DPC++FPGA/Tutorials/Tools/board_test/README.md

```diff
@@ -0,0 +1,346 @@
+
+`board_test` measures the frequency that the kernel is running at in the FPGA and compares this to compiled kernel clock frequency. [Test Details](#test-details) section provides more information.
+
+*Note:* The block diagram shown below is an overview of a typical oneAPI FPGA platform. Please refer to oneAPI platform/BSP vendor documentation for more details on platform components.
+
+![](oneapi_fpga_platform.PNG)
+
+### Test Details
+The complete test exercises following interfaces in a platform:
+1. **Host to device global memory interface:** This interface is checked by performing explicit data movement between the host and device global memory. Host to device global memory bandwidth is measured and reported. As a part of this interface check, unaligned data transfers are also performed to verify non-DMA transfers complete successfully.
+
+2. **Kernel to device global memory interface:** This interface is checked by performing kernel to memory data transfers using simple read and write kernels. Kernel to memory bandwidth is measured and reported.<br> 
+*Note:* This test currently does not support SYCL Unified Shared Memory (USM). For testing USM interface, please use [Simple host streaming sample](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2BFPGA/Tutorials/DesignPatterns/simple_host_streaming).
+
+3. **Host to kernel interface:** The test ensures the host to kernel communication is correct and that the host can launch a kernel successfully. It also measures the roundtrip kernel launch latency and throughput (number of kernels/ms) of single task no-operation kernels.
+
+In addition to the above interface checks, the test also measures the **Kernel clock frequency**. The test measures the frequency that the programmed kernel is running at on the FPGA device and reports it. By default, this test fails if the measured frequency is not within 2% of the compiled frequency.<br>
+*Note:* The test allows overriding this failure, however, this is may lead to functional errors and is not recommended. The override option is provided to allow debug in case where platform design changes are done to force kernel to run at slower clock (this is not a common use-case). Usage section describes how to override this option if required.
+
+The complete test is divided into six sub-tests. By default, all tests are run. You can choose to run a single test using `-test=<test number>` option. Please see [Running the Sample](#running-the-sample) for test usage instructions.
+1. Host Speed and Host Read Write Test
+2. Kernel Clock Frequency Test
+3. Kernel Launch Test
+4. Kernel Latency Measurement
+5. Kernel-to-Memory Read Write Test
+6. Kernel-to-Memory Bandwidth Test<br>
+*Note:* It is recommended to run all tests at least once to ensure that the platform interfaces are fully functional.
+
+## Key Concepts

```

Comment:
I think that you can remove this section as I feel it does not bring much information

Reply:
done